crates/remora/src/executor/fake.rs

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    cmp::Ordering,
+    collections::BTreeMap,
+    future::Future,
+    marker::PhantomData,
+    sync::{Arc, RwLock},
+    time::Duration,
+};
+
+use sui_types::{
+    base_types::{ObjectID, ObjectRef, SequenceNumber, SuiAddress},
+    committee::EpochId,
+    digests::{TransactionDigest, TransactionEventsDigest},
+    effects::{InputSharedObject, ObjectChange, TransactionEffectsAPI},
+    execution_status::ExecutionStatus,
+    gas::GasCostSummary,
+    object::{MoveObject, Object, Owner},
+    transaction::InputObjectKind,
+};
+use tokio::time::Instant;
+
+use super::api::{
+    ExecutableTransaction,
+    ExecutionResultsAndEffects,
+    Executor,
+    StateStore,
+    TransactionWithTimestamp,
+};
+
+/// A fake objects for testing.
+pub fn fake_owned_object(version: u64) -> Object {
+    let id = ObjectID::random();
+    let object_version = SequenceNumber::from_u64(version);
+    let owner = SuiAddress::random_for_testing_only();
+    Object::with_id_owner_version_for_testing(id, object_version, owner)
+}
+
+#[derive(Clone)]
+pub struct FakeTransaction {
+    pub digest: TransactionDigest,
+    inputs: Vec<InputObjectKind>,
+}
+
+impl FakeTransaction {
+    pub fn new(inputs: Vec<InputObjectKind>) -> Self {
```

Comment:
The inputs argument needs certain construction as shown in the test samples.
```rust
let inputs: Vec<_> = (0..2)
  .map(|_| {
      let object = fake_owned_object(0);
      let reference = object.compute_object_reference();
      store.write_object(object);
      InputObjectKind::ImmOrOwnedMoveObject(reference)
  })
  .collect();
let transaction = FakeTransaction::new(inputs);
```
Can we have a better interface (i.e., object id and the object type, owned/shared) or add a helper function for this?

Reply:
Good idea